src/coordinator.rs

```diff
@@ -78,29 +86,61 @@ impl<'job> Coordinator<'job> {
             )
             .context("could not run job")?;
 
-        for (blocked, blockers) in self.blocked.iter_mut() {
-            if blockers.remove(&next) && blockers.is_empty() {
-                self.ready.push(*blocked);
+        // Now that we're done running the job, we update our bookkeeping to
+        // figure out what running that job just unblocked.
+        //
+        // As an implementation note, this will probably end up in a separate
+        // function once we're running tasks in parallel!
+        let mut newly_unblocked = Vec::default(); // avoiding mutating both fields of self in the loop below
```

Comment:
super nit: I don't think I have every seen anyone use Vec::default(). I think it is much more common to use vec![].

Reply:
thanks! I haven't done a huge amount of Rust collaboratively—I'll change this to be less weird.